### ISSUE TITLE
hhi not found

### ISSUE THREAD

**Issue by avikchaudhuri** [facebook/flow#590]
Somebody ran into this weird issue running Flow on a laptop: running `bin/flow` from the flow installation root works, but running `flow` errors out with ".hhi not found." Changing the $PATH setting on the laptop that finds flow to use absolute instead of `~` paths seems to resolve the problem.

Are we making any shell-specific assumptions in the part of the code that embeds core libs? @mroch @gabelevi @samwgoldman

**Comment by gabelevi** [comment]
Well first of all that error message is [bad copypasta](https://github.com/facebook/flow/blob/444dbd08f0851f617b88ef00b141f2adc628ee59/src/common/files_js.ml#L49)

Second, it kind of sounds like Flow is failing to extract the embedded flowlibs. Perhaps this user was running without permissions for their tmp dir (possible stored in the $TMPDIR env var)?

It's really weird that changing the path affects anything though...

**Comment by mroch** [comment]
i suspect that we're missing a `realpath` somewhere in there

**Comment by mroch** [comment]
I can repro like this:

```
$ PATH='~/Code/flow/bin':/usr/bin flow
Could not locate hhi files
```

(`/usr/bin` is necessary for `tar` and `find`)

if I remove the quotes so that bash expands the `~`, it works fine.

**Comment by mroch** [comment]
Fixed by dde7725666a57c00a0b210f23dc75da85813e95b
